lib/org/chromium/webidl/BlinkIDLFileDAO.js

```diff
@@ -95,64 +120,108 @@ foam.CLASS({
+        it.`,
       code: function() {
-        var self = this;
-        return self.HTTPRequest.create({
-          url: self.BRANCH_TO_COMMIT_URL,
-          responseType: 'text', // Security prefix yields malformed JSON.
-        }).send().then(function(response) {
-          if (response.status !== 200) throw response;
-          return response.payload;
-        }).then(function(payload) {
-          var json = JSON.parse(payload.substr(4)); // Skip security prefix.
-          self.commit = json.commit;
-          return self;
+        var exists = this.fs_.existsSync;
+        if (exists(this.localRepositoryPath) &&
+            exists(this.path_.resolve(this.localRepositoryPath, '.git'))) {
+          return Promise.resolve();
+        }
+
+        // Invoke in shell:
+        // <fetch-script> <local-repo-path> <remote-repo-url> <sparse-path>
+        var child = this.childProcess_.execFile(
+            this.path_.resolve(__dirname, this.SCRIPT_RELATIVE_FETCH_PATH),
+            [this.localRepositoryPath, this.REPOSITORY_URL, this.SPARSE_PATH]);
+        child.stdout.pipe(this.process_.stdout);
+        child.stderr.pipe(this.process_.stderr);
+
+        return new Promise(function(resolve, reject) {
+          child.on('exit', function(status, signal) {
+            if (status === 0) {
+              resolve();
+            } else {
+              reject(new Error(`Non-zero exit status from git repository fetch
+                script: ${status}; signal: ${signal}`));
+            }
+          });
         });
       },
     },
   ],
 
   listeners: [
-    function onCommitSet() {
-      var self = this;
-      return self.HTTPRequest.create({
-        url: self.codeSearchQueryURL,
-        responseType: 'json',
-      }).send().then(function(response) {
-        if (response.status !== 200) throw response;
-        return response.payload;
-      }).then(function(json) {
-        if (json.search_response.length !== 1)
-          throw new Error('Unexpected multi-part search_response from Chromium Code Search');
-        var data = json.search_response[0];
-        if (data.maybe_skipped_documents)
-          throw new Error('Some Chromium Code Search results may have been skipped');
-        if (data.results_offset !== 0)
-          throw new Error('Chromium Code Search skipped some results');
-        if (data.estimated_total_number_of_results !== data.search_result.length)
-          throw new Error('Chromium Code Search result set incomplete');
+    {
+      name: 'onFetched',
+      documentation: `Handler for post-repository fetched. Update "commit" and
+        locate IDL files.`,
+      code: function() {
+        this.commit = this.childProcess_.execSync(
+            `"${this.GIT_PATH}" rev-parse HEAD`, {cwd: this.localRepositoryPath})
+                .toString().trim();
+        var execStr = `"${this.FIND_PATH}" "${this.localRepositoryPath}" -type f -path "${this.localRepositoryPath}/${this.SPARSE_PATH}/*.idl"`;
```

Comment:
Using -name "\*.idl" instead of -path "$vars/*.idl" might be a bit easier to understand.